tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/executable/cases/project/NewEspressifIDFProjectTest.java

```diff
@@ -417,5 +462,39 @@ private static void switchEditorToSourceIfPresent(SWTBotEditor editor)
 			}
 
 		}
+
+		private static void whenProjectCleanUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "ESP-IDF: Project Clean");
+			ProjectTestOperations.joinJobByName(Messages.ProjectCleanCommandHandler_RunningProjectCleanJobName);
+			ProjectTestOperations.waitForProjectClean(bot);
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "Refresh");
+		}
+
+		private static void whenProjectFullCleanUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "ESP-IDF: Project Full Clean");
+			ProjectTestOperations.joinJobByName(Messages.ProjectFullCleanCommandHandler_RunningFullcleanJobName);
+			ProjectTestOperations.waitForProjectClean(bot);
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "Refresh");
+		}
+
+		private static void whenProjectPythonCleanUsingContextMenu() throws IOException
+		{
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "ESP-IDF: Python Clean");
+			ProjectTestOperations.joinJobByName(Messages.PythonCleanCommandHandler_RunningPythonCleanJobName);
+			ProjectTestOperations.waitForProjectClean(bot);
+			ProjectTestOperations.launchCommandUsingContextMenu(projectName, bot, "Refresh");
+		}
+
+		private static void checkIfProjectCleanedFilesInBuildFolder() throws IOException
+		{
+			ProjectTestOperations.findProjectCleanedFilesInBuildFolder(projectName, bot);
+		}
+
+		private static void checkIfProjectFullCleanedFilesInBuildFolder() throws IOException
+		{
+			ProjectTestOperations.findProjectFullCleanedFilesInBuildFolder(projectName, bot);
+		}
```

Comment:
these should be assertions and as I left a comment on other methods this should be assertion on true or false